packages/typespec-test/test/versioning/generated/typespec-ts/src/v2/clientDefinitions.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  GetParameters,
+  UpdateParameters,
+  DeleteParameters,
+  CreateParameters,
+  ListParameters,
+  CustomGetParameters,
+} from "./parameters";
+import {
+  Get200Response,
+  GetDefaultResponse,
+  Update200Response,
+  UpdateDefaultResponse,
+  DeleteOperation200Response,
+  DeleteOperationDefaultResponse,
+  Create200Response,
+  Create201Response,
+  CreateDefaultResponse,
+  List200Response,
+  ListDefaultResponse,
+  CustomGet200Response,
+} from "./responses";
+import { Client, StreamableMethod } from "@azure-rest/core-client";
+
+export interface Get {
+  /** Gets an instance of the resource. */
+  get(
+    options?: GetParameters,
+  ): StreamableMethod<Get200Response | GetDefaultResponse>;
+  /** Updates an existing instance of the resource. */
+  patch(
+    options: UpdateParameters,
+  ): StreamableMethod<Update200Response | UpdateDefaultResponse>;
+  /** Deletes an existing instance of the resource. */
+  delete(
+    options?: DeleteParameters,
+  ): StreamableMethod<
+    DeleteOperation200Response | DeleteOperationDefaultResponse
+  >;
+}
+
+export interface Create {
+  /** Creates a new instance of the resource. */
+  post(
+    options: CreateParameters,
+  ): StreamableMethod<
+    Create200Response | Create201Response | CreateDefaultResponse
+  >;
+  /** Lists all instances of the resource. */
+  get(
+    options?: ListParameters,
+  ): StreamableMethod<List200Response | ListDefaultResponse>;
+}
+
+export interface CustomGet {
+  get(options?: CustomGetParameters): StreamableMethod<CustomGet200Response>;
+}
+
+export interface Routes {
+  /** Resource for '/\{id\}' has methods for the following verbs: get, patch, delete */
+  (path: "/{id}", id: string): Get;
+  /** Resource for '/' has methods for the following verbs: post, get */
+  (path: "/"): Create;
+  /** Resource for '/customGet' has methods for the following verbs: get */
+  (path: "/customGet"): CustomGet;
+}
+
+export type DemoServiceClient = Client & {
```

Comment:
If we don't add suffix v2 to the client name, will it be confusing to our customer ?

Reply:
I wouldn't like to alter the generated names. If a user just needs a single version which may be the most common scenario they won't really care to know if they are on DemoServiceClientv1 or v2 since they have already imported it.

```ts
import {v2} from "@azure-rest/foo"

const client: v2.DemoServiceClient = ...
```